cpp/benchmarks/io/csv/csv_reader_options.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+#include <benchmarks/io/cuio_common.hpp>
+#include <benchmarks/io/nvbench_helpers.hpp>
+
+#include <cudf/io/csv.hpp>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+constexpr size_t data_size = 256 << 20;
+
+template <column_selection ColSelection, row_selection RowSelection>
+void BM_csv_read_varying_options(
+  nvbench::state& state,
+  nvbench::type_list<nvbench::enum_type<ColSelection>, nvbench::enum_type<RowSelection>>)
+{
+  cudf::rmm_pool_raii rmm_pool;
+
+  auto constexpr num_chunks = 1;
```

Comment:
Comments on why is this?

Reply:
@vuule Do we want benchmarks varying `num_chunks` in this case?

Reply:
We should use different numbers of chunks in the "csv_read_row_selection" benchmark. The old code used 1 and 8, we keep it that way for now.

Reply:
Done